core/blocks.py

```
@@ -242,21 +240,49 @@ class Meta:
-
-        queryset = models.CaseStudy.objects.all()
-        for filter_arg in filter_args:
-            cs_queryset = queryset.filter(**filter_arg)
-            if cs_queryset.exists():
-                context['case_study'] = get_most_ranked_case_study(cs_queryset=cs_queryset, context=context)
-                break
+        try:
+            s = search(
+                export_commodity_codes=export_commodity_codes,
+                export_markets=export_markets,
+                export_regions=export_regions,
+                page_context=page_context,
+            )
+            if s.count():
+                from core.models import CaseStudyScoringSettings
+
+                settings = CaseStudyScoringSettings.for_request(context['request'])
+                hits = []
+                for hit in s.scan():
+                    hit_dict = hit.to_dict()
+                    hit_dict['score'] = get_cs_ranking(
+                        hit_dict,
+                        export_commodity_codes=export_commodity_codes,
+                        export_markets=export_markets,
+                        export_regions=export_regions,
+                        page_context=page_context,
+                        export_blocs=export_blocs,
+                        settings=settings,
+                    )
+                    hits.append(hit_dict)
+                sorted_cs_list = sorted(hits, key=lambda hit: hit.get('score'), reverse=True)
+                best_case_study = sorted_cs_list and sorted_cs_list[0]
+                if best_case_study and int(best_case_study.get('score')) >= getattr(settings, 'threshold'):
+                    best_case_study_id = best_case_study.get('pk')
+                    context['case_study'] = models.CaseStudy.objects.get(id=best_case_study_id)
+        except ConnectionError:
+            # nothing we can do without elasticsearch so continue without case study
```

can see why this is here but i'm not super keen on swallowing exceptions. What happens if we got bad credentials we won't know until issue is reported. May be at very least log the error so sentry picks this up.

---

OK - good plan. Yeah - this is present only so the page doesn't break if elastic-search is not available.